### DISCUSSION TITLE
bug (?) Is there a product limit of 1000 on Storefront API graphql queries?

### DISCUSSION THREAD

**Discussion by johansonzap** [Shopify/storefront-api-feedback#115] [Help]
We are fetching our products through a graphQL query via Storefront API.
Our collection has over 1000 products.
If I fetch the products in that collection, it will give me the expected response(s) of the 3000 or so products, if i page through them. 
However, if i fetch the products in that collection using the "filters" param, it will limit the results to 1000 products (4 pages of 250 products each), even though there are over 1000 products that meet that filters criteria.  Various filters have this same result.

Example query with the filters param below.  If i remove the filters param, i get the correct number of results.  With the filters param, it will not return more than 1000 results

Can anyone confirm there is now a limit of 1000 products on the Storefront API graphql responses when filtered?  
This behavior seemed to have changed in the last couple of weeks (not exactly sure when).

Example Query
            query getProductCursors{
              collection(handle: "all-products") {
                products(first: 250, sortKey: BEST_SELLING, filters:[{available:true}]) {
                    edges{
                      cursor
                    }
                    pageInfo{
                      hasNextPage
                      endCursor
                    }
                }
              }
            }

**Comment by patjakubik** [comment]
Yes there is a limit, and there is no plan to increase it in near future (source: Partners Slack)

**Reply by johansonzap** [reply]
Thank you for your response.  Just to clarify...

So the **storefront API** with no filters does not have a limit (it returns over 3000 products).
But the **storefront API** with using the "filters=" param does have a limit of 1000 products?

Please note I am not referencing the **Admin API**.

**Reply by patjakubik** [reply]
Both Storefront API and Liquid Online Store can only use filters in collections that have maximum of 1000 products. 

https://help.shopify.com/en/manual/online-store/themes/customizing-themes/storefront-filters#:~:text=If%20your%20product%20collection%20contains%20over%201000%20products%2C%20then%20filters%20are%20hidden.

**Reply by johansonzap** [reply]
Thank you for your help.  I must have thought it worked differently.

Its odd that the filtering "works" on the collection of more than 1k, it just stops paginating at 1000 products.
If I have a collection of say 3k products, submit a filter that would bring back 2k, it brings back the first 1k.  Filters can be added to eventually get that under 1k, at which time all products are shown and are filtered correctly.  Thus the filtering is working on the collection, the response just doesn't return all of the results.

Still better than just an empty filter drawer...
Again, thanks for confirming.

**Reply by patjakubik** [reply]
Yeah it's a pain 🤷, we ended using Algolia...

**Comment by johncraigcole** [comment]
Hey everyone 👋 . I know this is an old thread, but wanted to share that the filtering on products within a collection is now on the first 5K. We're still wanting to push this much further so while I don't have a timeline keep an eye out for improvements coming in the future.
